samlsp/middleware.go

```diff
@@ -44,93 +39,84 @@ import (
 // SAML service provider already has a private key, we borrow that key
 // to sign the JWTs as well.
 type Middleware struct {
-	ServiceProvider   saml.ServiceProvider
-	AllowIDPInitiated bool
-	TokenMaxAge       time.Duration
-	ClientState       ClientState
-	ClientToken       ClientToken
-	Binding           string
-}
-
-var (
-	jwtSigningMethod = jwt.SigningMethodRS256
-	jwtParser        = jwt.Parser{
-		ValidMethods: []string{jwtSigningMethod.Name},
-	}
-)
-
-func randomBytes(n int) []byte {
-	rv := make([]byte, n)
-	if _, err := io.ReadFull(saml.RandReader, rv); err != nil {
-		panic(err)
-	}
-	return rv
-}
-
-func (m *Middleware) issueSignedToken(claims jwt.Claims) (string, error) {
-	token := jwt.NewWithClaims(jwtSigningMethod, claims)
-	return token.SignedString(m.ServiceProvider.Key)
-}
-
-func (m *Middleware) parseSignedToken(token string) (*jwt.Token, error) {
-	return jwtParser.Parse(token, func(*jwt.Token) (interface{}, error) {
-		return m.ServiceProvider.Key.Public(), nil
-	})
-}
-
-func (m *Middleware) parseSignedTokenWithClaims(token string, claims jwt.Claims) (*jwt.Token, error) {
-	return jwtParser.ParseWithClaims(token, claims, func(*jwt.Token) (interface{}, error) {
-		return m.ServiceProvider.Key.Public(), nil
-	})
+	ServiceProvider saml.ServiceProvider
+	OnError         func(w http.ResponseWriter, r *http.Request, err error)
+	Binding         string // either saml.HTTPPostBinding or saml.HTTPRedirectBinding
```

Comment:
just a thought, why is saml.HTTPPostBinding a `var` not a `const`?

Reply:
I'm not sure, but there must have been a reason. I know there is/was in issue with taking the address of a string constant, so that may have been it. Things compile like fine with const, so I'll include that.